lib/IRGen/LoadableByAddress.cpp

```diff
@@ -942,11 +942,25 @@ static SILInstruction *createOutlinedCopyCall(SILBuilder &copyBuilder,
   return copy;
 }
 
+// Find the correct debug scope for alloc stack. We want to give to the
+// expanded sequence the correct debug scope so we skip over instructions
+// that aren't lowered to anything real (e.g. debug_value).
+static const SILDebugScope *findAllocStackDebugScope(SILBasicBlock &BB) {
+  auto It = BB.begin();
+  while (It != BB.end()) {
+    if (!isMaintenanceInst(&*It))
+      return It->getDebugScope();
+    It++;
+  }
+  return BB.begin()->getDebugScope();
+}
+
 void LoadableStorageAllocation::replaceLoadWithCopyAddr(
     LoadInst *optimizableLoad) {
   SILValue value = optimizableLoad->getOperand();
 
-  SILBuilderWithScope allocBuilder(pass.F->begin()->begin());
+  SILBuilder allocBuilder(pass.F->begin()->begin());
+  allocBuilder.setCurrentDebugScope(findAllocStackDebugScope(*pass.F->begin()));
```

Comment:
Wouldn't it be better to add a new constructor to SILBuilderWithScope that has this behavior?

```
/// Creates a new SILBuilder with an insertion point at the beginning of BB and the debug scope from the first non-metainstruction in the BB.
explicit SILBuilderWithScope(SILBasicBlock *BB) {}
```